src/System.Numerics.Tensors/System/Numerics/Tensor.cs

```diff
@@ -1042,12 +1020,23 @@ private bool Equals(Tensor<T> other, IEqualityComparer comparer)
 
             for (int i = 0; i < Length; i++)
             {
-                ArrayUtilities.GetIndices(strides, IsReversedStride, i, indices);
-
-                if (!comparer.Equals(this[indices], other[indices]))
+                if (IsReversedStride == other.IsReversedStride)
```

Comment:
(minor) would it make sense to move this branch out of the loop?  Checking it each time in the loop seems unnecessary.

Reply:
Yeah, I considered it but opted not to duplicate the loop.  I was hoping that the compiler would be smart enough to lift it since both of those are backed by readonly bools, but maybe not.

Reply:
I went ahead and lifted it out, did the same for CompareTo